src/main/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileup.scala

```diff
@@ -182,14 +182,14 @@ object AggregateSvPileup {
    *
    * @param pileup The pileup representative of the cluster that will be returned
    * @param pileupToNeighbors Map of pileup to its set of neighbors
+   * @parma neighborsToIgnore Set of pileups to ignore, as they are being considered in parent calls to this method
    * @return (1) The complete cluster for the given pileup; (2) Reduced version of `pileupToNeighbors` with all pileups
    *         in the returned cluster having been removed from both keys and values.
    */
-  def extractClusterFor(pileup: BreakpointPileup, pileupToNeighbors: Map[BreakpointPileup, PileupGroup]):
+  def extractClusterFor(pileup: BreakpointPileup, pileupToNeighbors: Map[BreakpointPileup, PileupGroup], neighborsToIgnore: Set[BreakpointPileup]):
   (PileupGroup, Map[BreakpointPileup, PileupGroup]) = {
-
     // Identify the immediate neighbors of the given pileup, and start a cluster containing them
-    val neighbors: PileupGroup = pileupToNeighbors.getOrElse(pileup, IndexedSeq())
+    val neighbors: PileupGroup = pileupToNeighbors.getOrElse(pileup, IndexedSeq()).filter(neighborsToIgnore.contains)
```

Comment:
this needs careful thought on correctness

Reply:
@nh13 are you able to generate an example that causes a stack overflow and add it as a test case?

At a minimum, we should fix the compile errors in the [test file](https://github.com/fulcrumgenomics/fgsv/blob/main/src/test/scala/com/fulcrumgenomics/sv/tools/AggregateSvPileupTest.scala) and make sure we haven't broken any known cases.